02_knight_and_princess/anton_potapov/graph_create.cpp

```diff
@@ -0,0 +1,115 @@
+#include <cassert>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "properties.hpp"
+
+using VertexId = int;
+using EdgeId = int;
+
+struct Vertex {
+  VertexId id;
+  Camp camp_info;
+  Vertex(int vertex_id, Camp c_info = Camp())
+      : id(vertex_id), camp_info(c_info) {}
+};
+
+struct Edge {
+  EdgeId id;
+  VertexId vertex1_id, vertex2_id;
+  Boss boss_info;
+  Edge(EdgeId e_id, VertexId ver1, VertexId ver2, Boss b_info = Boss())
+      : id(e_id), vertex1_id(ver1), vertex2_id(ver2), boss_info(b_info) {}
+};
+
+class Graph {
+ public:
+  Graph() = default;
+  Graph(const std::vector<Vertex>& vertices, const std::vector<Edge>& edges)
+      : vertices_(vertices), edges_(edges) {
+    for (const auto& edge : edges) {
+      adjacency_list_[edge.vertex1_id].push_back(edge.id);
+      adjacency_list_[edge.vertex2_id].push_back(edge.id);
+    }
+  }
+  std::string get_json_string() const {
+    std::stringstream json_stringstream;
+    json_stringstream << "{\n";
+
+    json_stringstream << "\t\"vertices\": [\n";
+    for (auto it = adjacency_list_.begin(); it != adjacency_list_.end(); ++it) {
+      const auto& [ith_vertex_id, ith_adjacency_list] = *it;
+      json_stringstream << "\t\t{\n";
+      json_stringstream << "\t\t\t\"id\": " << ith_vertex_id << ",\n";
+      json_stringstream << "\t\t\t\"edge_ids\": [";
+      for (size_t j = 0; j < ith_adjacency_list.size(); ++j) {
+        json_stringstream << ith_adjacency_list[j];
+        if (j + 1 != ith_adjacency_list.size()) {
+          json_stringstream << ", ";
+        }
+      }
+      json_stringstream << "]\n";
+      json_stringstream << "\t\t}";
+      if (std::next(it) == adjacency_list_.end()) {
+        json_stringstream << "\n";
```

Comment:
`std::endl` - кроссплатформенная версия новой строки
```suggestion
        json_stringstream << std::endl;
```